src/txmempool.cpp

```diff
@@ -212,6 +244,7 @@ void CTxMemPool::removeForBlock(const std::vector<CTransaction>& vtx, unsigned i
     BOOST_FOREACH(const CTransaction& tx, vtx)
     {
         std::list<CTransaction> dummy;
+        updateDependentPriorities(tx, nBlockHeight, true);
```

Comment:
I think you need to pass in a queue of transactions to update, and call this function once for the whole block.